src/cmd/ksh93/tests/sh_match.sh

```diff
@@ -411,12 +411,131 @@ function test_nomatch
     [[ $got == "$expect" ]] || log_error 'unset submatches not handled correctly'
 }
 
-# run tests
+# ========
+# This is a regexp to extract pattern to pattern entries from related API unit test source. It emits
+# the patterns in a form this script can easily use; i.e., the two patterns separated by a tab.
+# See the next two test functions.
+p2p_sep=$'\t'
+p2p_re=$'/ tests\\[\\] =/,/NULL, NULL/s/^.*{"\\(.*\\)", "\\(.*\\)"},$/\\1\t\\2/p'
+# These patterns are known not to be correctly converted to a glob. We still test them but only
+# issue a warning rather than failing the test. See https://github.com/att/ast/issues/1367.
+typeset -A ere2glob_blacklist
```

Comment:
Definition of `ere2glob_blacklist` and `glob2ere_blacklist` can go in respective function definitions.

Reply:
I originally had them in their respective functions. I moved them here to emphasize how the two sets are related to each other. Bit of a coin toss as to which location is better.